pkg/gatewaysecret/handler.go

```diff
@@ -0,0 +1,182 @@
+package gatewaysecret
```

Comment:
There are no unit tests in this package.
As we try to increase our code quality, it would be nice to provide some tests here.
A hint for what can be tested: I'd include the secret-watching logic, all time comparisons here and also that the "lastModifiedAt" annotation is always present on the secret.
Remember that the unit tests should not just be written to "test a scenario": They should be written with an intention to document the expected behavior of the unit (in our case, the unit is usually a single function/method)

Reply:
I have added extensive unit tests for all the functions in `handler.go`. The struct now has a dependency injection to make it more testable. PTAL since I am not sure if my testing and mocking approach is a standard one.